stdlib/importlib/abc.pyi

```diff
@@ -82,24 +91,82 @@ if sys.version_info >= (3, 7):
+        def is_dir(self) -> bool: ...
         @abstractmethod
-        def read_bytes(self) -> bytes: ...
+        def is_file(self) -> bool: ...
         @abstractmethod
-        def read_text(self, encoding: Optional[str] = ...) -> str: ...
+        def iterdir(self: _P) -> Generator[_P, None, None]: ...
         @abstractmethod
-        def is_dir(self) -> bool: ...
+        def joinpath(self: _P, child: StrPath) -> _P: ...
+        # Adapted from builtins.open
+        # Text mode: always returns a TextIOWrapper
+        @overload
         @abstractmethod
-        def is_file(self) -> bool: ...
+        def open(
+            self,
+            mode: OpenTextMode = ...,
+            buffering: int = ...,
+            encoding: str | None = ...,
+            errors: str | None = ...,
+            newline: str | None = ...,
+        ) -> TextIOWrapper: ...
+        # Unbuffered binary mode: returns a FileIO
+        @overload
+        @abstractmethod
+        def open(
+            self, mode: OpenBinaryMode, buffering: Literal[0], encoding: None = ..., errors: None = ..., newline: None = ...
+        ) -> FileIO: ...
+        # Buffering is on: return BufferedRandom, BufferedReader, or BufferedWriter
+        @overload
         @abstractmethod
-        def joinpath(self, child: _Path) -> Traversable: ...
+        def open(
+            self,
+            mode: OpenBinaryModeUpdating,
+            buffering: Literal[-1, 1] = ...,
+            encoding: None = ...,
+            errors: None = ...,
+            newline: None = ...,
+        ) -> BufferedRandom: ...
+        @overload
         @abstractmethod
-        def __truediv__(self, child: _Path) -> Traversable: ...
+        def open(
+            self,
+            mode: OpenBinaryModeWriting,
+            buffering: Literal[-1, 1] = ...,
+            encoding: None = ...,
+            errors: None = ...,
+            newline: None = ...,
+        ) -> BufferedWriter: ...
+        @overload
         @abstractmethod
-        def open(self, mode: Literal["r", "rb"] = ..., *args: Any, **kwargs: Any) -> IO[Any]: ...
-        @property
+        def open(
+            self,
+            mode: OpenBinaryModeReading,
+            buffering: Literal[-1, 1] = ...,
+            encoding: None = ...,
+            errors: None = ...,
+            newline: None = ...,
+        ) -> BufferedReader: ...
+        # Buffering cannot be determined: fall back to BinaryIO
+        @overload
+        @abstractmethod
+        def open(
+            self, mode: OpenBinaryMode, buffering: int, encoding: None = ..., errors: None = ..., newline: None = ...
+        ) -> BinaryIO: ...
+        # Fallback if mode is not specified
+        @overload
+        @abstractmethod
+        def open(
+            self, mode: str, buffering: int = ..., encoding: str | None = ..., errors: str | None = ..., newline: str | None = ...
+        ) -> IO[Any]: ...
+        name: str
```

Comment:
This is more restrictive than the previous `@property def name(self) -> str: ...`, because it requires `any_traversible.name = "Lol"` to work, whereas a property without a setter only requires a readable name attribute.

Reply:
Changed!